solana/pyth2wormhole/client/src/batch_state.rs

```diff
@@ -40,63 +43,46 @@ impl<'a> BatchState<'a> {
             symbols,
             conditions,
             last_known_symbol_states: vec![None; symbols.len()],
-            status: BatchTxStatus::Sending { attempt_no: 1 },
-            status_changed_at: Instant::now(),
-            last_success_at: None,
+            last_job_finished_at: Instant::now(),
         }
     }
-    /// Ensure only set_status() alters the timestamp
-    pub fn get_status_changed_at(&self) -> &Instant {
-        &self.status_changed_at
-    }
-    pub fn get_status(&self) -> &BatchTxStatus {
-        &self.status
-    }
-
-    /// Ensure that status changes are accompanied by a timestamp bump
-    pub fn set_status(&mut self, s: BatchTxStatus) {
-        self.status_changed_at = Instant::now();
-        self.status = s;
-    }
 
     /// Evaluate the configured attestation conditions for this
     /// batch. RPC is used to update last known state. Returns
     /// Some("<reason>") if any trigger condition was met. Only the
     /// first encountered condition is mentioned.
-    pub fn should_resend(&mut self, c: &RpcClient) -> Option<String> {
+    pub async fn should_resend(&mut self, c: &RpcClient) -> Option<String> {
         let mut ret = None;
 
         let sym_count = self.symbols.len();
-        let mut new_symbol_states: Vec<Option<PriceAccount>> = Vec::with_capacity(sym_count);
-        for (idx, sym) in self.symbols.iter().enumerate() {
-            let new_state = match c
-                .get_account_data(&sym.price_addr)
-                .map_err(|e| e.to_string())
-                .and_then(|bytes| {
-                    pyth_sdk_solana::state::load_price_account(&bytes)
-                        .map(|state| state.clone())
-                        .map_err(|e| e.to_string())
-                }) {
-                Ok(state) => Some(state),
-                Err(e) => {
-                    warn!(
-                        "Symbol {} ({}/{}): Could not look up state: {}",
-                        sym.name
-                            .as_ref()
-                            .unwrap_or(&format!("Unnamed product {}", sym.product_addr)),
-                        idx + 1,
-                        sym_count,
-                        e.to_string()
-                    );
-                    None
-                }
-            };
-
-            new_symbol_states.push(new_state);
+        let mut new_symbol_states: Vec<Option<PriceAccount>> = vec![None; sym_count];
+        let pubkeys: Vec<_> = self.symbols.iter().map(|s| s.price_addr).collect();
+
+        // Always learn the current on-chain state for each symbol
+        match c.get_multiple_accounts(&pubkeys).await {
+            Ok(acc_opts) => {
```

Comment:
Changing `new_symbol_states` to be a mapping of `Pubkey` to `PriceAccount`s might let you simplify this logic.

Reply:
This does seem nice, but a map will obfuscate the order in which the symbols were configured (which could be useful for cases where we want to rule out typos in failing batch configs)